### PR TITLE
pr of irumeria

### PR DESCRIPTION
**Pull request by irumeria**
Totally 4 places are changed in this pr:

1. the usage of command line parser "clap" is updated with its version 4 features.
2. add a test for examining the rotation invariance of soap (in main.rs, function "test_rotate") 
3. (Important) The parameters "theta" and "phi" are exchanged in the function "to_polar". Since I think in the way you write the function "sphr_harm", "theta" should be the one that towards the positive Z axis. (Note that this exchange reduce the loss in the rotation invariance test, but the loss is still some kind of large)
4. Multiple neighbour elements can be selected now, by arg "--species" or "-s"

### CODE REVIEW

**Comment by t-young31** [comment]
hi @irumeria – thanks a lot for the PR. will take a look this weekend 

(oh and top GitHub tip: if you want to add more commits to this PR you can just push to your `iru` branch rather than closing it and reopening another!)

**Comment by irumeria** [comment]
Oh, thank you for the tip.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 
 target/
-
+*.xyz
 Cargo.lock
+.vscode
```
Comment (t-young31) [inline .gitignore]: needs a newline at the end of the file

File: Cargo.toml [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+clap = {version = "^4", features=["derive"]}
 rand = "*"
 GSL = "*"
 ndarray = "*"
+
+[profile.release]
+lto = true
+codegen-units = 1
+strip = true
+opt-level = "s"
```
Comment (t-young31) [inline Cargo.toml]: why opt for binary size? Would have thought turning all optimisations would be best..?

File: README.md [hunk 3]
```diff
@@ -18,10 +18,10 @@ cargo build --release
 
 
 ##  Usage  
-Execute the binary built in the *targets/release/* directory. For example to generate the 'SOAP vector' for the H-atom density in a methane molecule:
+Execute the binary built in the *targets/release/* directory. For example to generate the 'SOAP vector' for the density of all of the Na and Cl in a Nacl lattice:
 
 ```bash
-target/release/soap methane.xyz 0 H
+target/release/soap -f=./nacl_test.xyz -c=0 -s Na -s  Cl
```
Comment (t-young31) [inline README.md]: I'm not a fan of having e.g. `-c=0` CLIs are usually `-c 0` is this behaviour supported?

File: src/main.rs [hunk 4]
```diff
@@ -1,72 +1,136 @@
...
+            6.,
+            0.5,
+        );
+        let sum1:f64 = p.iter().map(|x|x.abs()).sum();
+        let sum2:f64 = p2.iter().map(|x|x.abs()).sum();
+        let sum:f64 = p.iter().zip(p2.iter()).map(|(x1,x2)|(x1-x2).abs()).sum();
+        println!("sum of soap before ratation p1: {:?}",sum1);
+        println!("sum of soap after ratation p2: {:?}",sum2);
+        println!("sum of (p1 - p2): {:?}",sum);
+
```
Comment (t-young31) [inline src/main.rs:131]: tests need to assert something!

File: src/soap.rs [hunk 5]
```diff
@@ -183,7 +187,91 @@ mod tests{
         (x - y).abs() <= atol
     }
 
-    
+    pub fn power_spectrum_old(structure:      &Structure,
```
Comment (t-young31) [inline src/soap.rs]: not a fan of having an `<X>_old` function – worth reconciling

**Review by t-young31 — CHANGES_REQUESTED** [review]
thanks again for the PR @irumeria – I've left a few comments but what's more critical is that the tests now no longer pass (assuming due to the theta<->phi swap). I've not looked into what the problem is there so any work on that would be appreciated!

**Comment by irumeria** [comment]
Thanks for the helpful comments.

The direct cause for test failing is the pre-computed values of Monte Carlo integral do not match the analytic result whose theta and phi are swapped. It passed again after updating  the pre-computed values. I will recheck the comments and push to my branch later.

**Review by t-young31 — APPROVED** [review]
Looks great – thanks! 😄

**Pull request merged**